homeassistant/components/mqtt/discovery.py

```diff
@@ -57,6 +137,32 @@
             _LOGGER.warning("Component %s is not supported", component)
             return
 
+        if payload:
+            try:
+                payload = json.loads(payload)
+            except ValueError:
+                _LOGGER.warning("Unable to parse JSON %s: '%s'",
+                                object_id, payload)
+                return
+
+        payload = dict(payload)
+
+        for key in list(payload.keys()):
+            abbreviated_key = key
+            tmp = key.split('_')
```

Comment:
We should not do this. We should explicitly map every key we want to translate. No splits and partial mappings.

Reply:
OK, how about now then?